scripts/new_project.kts

```diff
@@ -1,25 +1,72 @@
+#!/usr/bin/env kscript
+
+@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.1")
+@file:MavenRepository("central", "https://repo.maven.apache.org/maven2/")
+
+import kotlinx.coroutines.*
+import kotlin.coroutines.coroutineContext
+import java.io.File
+
 object NewProject {
 
+    private const val ARGUMENT_DELIMITER = "="
+    private const val COROUTINE_FOLDER_NAME = "CoroutineTemplate"
     private const val KEY_APP_NAME = "app-name"
     private const val KEY_PACKAGE_NAME = "package-name"
-    private const val ARGUMENT_DELIMITER = "="
+    private const val TEMPLATE_FOLDER_NAME = "coroutine"
 
     private var appName = ""
+    private var appNameWithoutSpace = ""
     private var packageName = ""
 
-    fun generate(args: Array<String>) {
+    fun generate(args: Array<String>) = runBlocking {
         handleArguments(args)
+
+        initializeNewProjectDirectory()
+    }
+
+    private suspend fun initializeNewProjectDirectory() {
+        showMessage("=> 🐢 Initializing new project...")
+        copyFiles(toPath = appNameWithoutSpace)
+        /*
+         * Set gradlew file as executable
+         * Copying file from one folder to another lost it's previous execution right.
+         * Need to set gradlew as executable to run
+         */
+        File("../$appNameWithoutSpace/gradlew")?.setExecutable(true)
     }
 
     private fun handleArguments(args: Array<String>) {
         args.forEach {
             val (key, value) = it.split(ARGUMENT_DELIMITER)
             when (key) {
-                KEY_APP_NAME -> appName = value
+                KEY_APP_NAME -> {
+                    appName = value
+                    appNameWithoutSpace = appName.trim().replace(" ", "")
+                }
                 KEY_PACKAGE_NAME -> packageName = value
             }
         }
     }
+
+    private fun showMessage(message: String?) {
+        runBlocking {
+            println("\n${message?.trimIndent()}\n")
+        }
+    }
+
+    private suspend fun copyFiles(toPath: String) {
```

Comment:
Should we refactor it to `copyFiles(fromPath: String, toPath: String)`? It will be more flexible this way

`copyFiles(fromPath = "../$COROUTINE_FOLDER_NAME", toPath = "../$appNameWithoutSpace")`